### PR TITLE
core: add support for MK66F18 (Teensy 3.6) micro controller

### PR DESCRIPTION
**Pull request by stapelberg**
This is in preparation for https://github.com/qmk/qmk_firmware/pull/10171

## Types of Changes

- [x] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* https://github.com/qmk/qmk_firmware/pull/10171

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: tmk_core/common/chibios/pin_defs.h [hunk 1]
```diff
@@ -53,6 +53,9 @@
 #        define D5 PAL_LINE(GPIOC, 13)
 #        define B0 PAL_LINE(GPIOC, 14)
 #    endif
+#elif defined(KEYBOARD_kinesis_kint36)
+// The compatibility defines do not work with the KINETIS header files,
+// so do not declare anything.
```
Comment (fauxpark) [inline tmk_core/common/chibios/pin_defs.h:58]:
```suggestion
```

File: quantum/mcu_selection.mk [hunk 2]
```diff
@@ -345,6 +345,33 @@ ifneq ($(findstring STM32G474, $(MCU)),)
   DFU_SUFFIX_ARGS ?= -v 0483 -p DF11
 endif
 
+ifneq ($(findstring MK66F18, $(MCU)),)
```
Comment (fauxpark) [inline quantum/mcu_selection.mk:348]: Move this up under the MK20DX256.

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: quantum/mcu_selection.mk (hunk 2 above)
Comment (stapelberg) [reply quantum/mcu_selection.mk:348]: Done.

File: tmk_core/common/chibios/pin_defs.h (hunk 1 above)
Comment (stapelberg) [reply tmk_core/common/chibios/pin_defs.h]: I’m a little confused about this suggestion—just removing these lines will result in a compilation error down the line (likely only once #10171 is applied).

Are you saying we should defer this change until later?

Comment (fauxpark) [reply tmk_core/common/chibios/pin_defs.h]: There needs to be a better solution than just hardcoding keyboard names into core code whenever someone wants to add a Teensy 3.6-based board.

What is the error it's producing?

Comment (stapelberg) [reply tmk_core/common/chibios/pin_defs.h]: > There needs to be a better solution than just hardcoding keyboard names into core code whenever someone wants to add a Teensy 3.6-based board.

Definitely agreed. If you can tell me how to make a MK66F18 define work, that’d be great!

> What is the error it's producing?

Here’s the full compilation error: https://gist.github.com/stapelberg/260314ef7f6310d3c54527c73e6442b0

Comment (fauxpark) [reply tmk_core/common/chibios/pin_defs.h]: I think this is due to the config_common.h include in the top-level config.h - this is also why the Input Club boards don't include that file. Considering the top-level one doesn't reference any of the pin definitions, moving the include to each of the sub-config.h's would probably "solve" it.

Comment (stapelberg) [reply tmk_core/common/chibios/pin_defs.h]: Ah, thanks for figuring this out! Yes, I think you are right.

We can change the config includes in #10171

**Comment by stapelberg** [comment]
This should be ready to merge now? :)

**Review by fauxpark — APPROVED** [review]

**Review by tzarc — APPROVED** [review]

**Comment by stapelberg** [comment]
Thank you!

**Pull request merged**